### PR TITLE
Test: Drop whitelist config for server and test folders

### PR DESCRIPTION
**Pull request by gziolo**
Part of #4596.

At the moment I removed `test.json` whitelist for `server` and `test` folder. We need to keep whitelist for `client` folder until we migrate all remaining test folders. To make everything compatible I also changes the way we enabled whitelist. It's now always enabled for `client` folder only.

I also renamed npm script for `test` folder from `test-tests` to `test-test` to align it with other subfolders.

### Testing
All test runs should pass:
* `npm run test-client`
* `npm run test-server`
* `npm run test-test`

Also for single test file:
* `npm run test-client client/lib/accept/test/index.js`
* `npm run test-test test/test/helpers/use-nock/test/index.js`

And finally we need to make sure that on Circle CI test runner doesn't run any tests if no test files are provided. Locally it should always load all possible tests when no test files are specified in npm command.

### CODE REVIEW

**Comment by gziolo** [comment]
I guess we can already take similar step for `server` folder, but I noticed there is one test that wasn't whitelisted which executes only when ENV is set for desktop. Which is very strange...

**Comment by gziolo** [comment]
Hmm, client test failed. No idea why, I'll look into that tomorrow ;)

**Comment by gziolo** [comment]
Circle CI is green again, but I have no idea why those tests failed before. I couldn't reproduce locally.

I dropped now also `tests.json` whitelist from `server` folder.

File: server/api/test/index.js [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+		localRequest
+			.get( '/logout' )
+			.redirects( 0 )
+			.set( 'cookie', 'wpcom_token=test' )
+			.expect( 'location', config( 'login_url' ) )
+			.expect( 'set-cookie', 'wpcom_token=; Path=/; Expires=Thu, 01 Jan 1970 00:00:00 GMT' )
+			.expect( 302, done );
+	} );
+
+	if ( process.env.NODE_ENV === 'desktop' ) {
```
Comment (gziolo) [inline server/api/test/index.js]: This one is tricky. We need to keep this check unless we mock all desktop specific config settings and whatever else we need to run it in `test` environment.

Comment (gziolo) [inline server/api/test/index.js]: I figured out there is open for this test #2111.

@mjangda: Any ideas what we should do about this one? We are using completely independent `test` env for all tests now. We can use test doubles for config or even update config file directly to make it work. Maybe we should even split this task into 2 PRs to make the process more transparent.

**Comment by gziolo** [comment]
@aduth @blowery: I added some tweaks which allow us to merge this PR right away :)

File: server/api/test/test.js [hunk 2]
```diff
@@ -6,23 +6,23 @@ var expect = require( 'chai' ).expect,
...
-	app = require( '../' )(),
-	localRequest = supertest( app );
-
 if ( process.env.NODE_ENV === 'desktop' ) {
 	describe( 'api', function() {
-		var sandbox = sinon.sandbox.create();
+		const sandbox = sinon.sandbox.create();
+		let app, config, localRequest;
+
+		before( () => {
```
Comment (gziolo) [inline server/api/test/test.js]: I prefer to be on a safe side and require all stuff here. I opened follow up to refactor this very `desktop` dependent test.

File: circle.yml [hunk 3]
```diff
@@ -15,17 +15,17 @@ test:
...
           - client/**/test/*.js
           - client/**/test/*.jsx
-    - NODE_ENV=test make client/config/index.js && MOCHA_FILE=./test-results-server.xml npm run test-server -- --reporter=mocha-junit-reporter -w:
+    - NODE_ENV=test make client/config/index.js && MOCHA_FILE=./test-results-server.xml npm run test-server -- --reporter=mocha-junit-reporter:
         parallel: true
         files:
           - server/**/test/*.js
           - server/**/test/*.jsx
-    - NODE_ENV=test make client/config/index.js && MOCHA_FILE=./test-results-tests.xml npm run test-tests -- --reporter=mocha-junit-reporter -w:
+    - NODE_ENV=test make client/config/index.js && MOCHA_FILE=./test-results-test.xml npm run test-test -- --reporter=mocha-junit-reporter:
```
Comment (blowery) [inline circle.yml]: can we make the building of the config an earlier part of the run and stop repeating it? 

I feel like I tried it and circle did some weird things.. Might be worth trying again to DRY this up.

Comment (gziolo) [inline circle.yml]: I'll give it a try :)

**Comment by gziolo** [comment]
I moved config generation to `pre` block and it works properly :tada:

**Comment by blowery** [comment]
🚢

**Pull request merged**